programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -112,7 +109,65 @@ pub fn deposit_v3(
         deposit_id: state.number_of_deposits,
         quote_timestamp,
         fill_deadline,
-        exclusivity_deadline,
+        exclusivity_deadline: current_time + exclusivity_period,
+        depositor,
+        recipient,
+        exclusive_relayer,
+        message,
+    });
+
+    Ok(())
+}
+
+pub fn deposit_v3_now(
+    ctx: Context<DepositV3>,
+    depositor: Pubkey,
+    recipient: Pubkey,
+    input_token: Pubkey,
+    output_token: Pubkey,
+    input_amount: u64,
+    output_amount: u64,
+    destination_chain_id: u64,
+    exclusive_relayer: Pubkey,
+    fill_deadline_offset: u32,
+    exclusivity_period: u32,
+    message: Vec<u8>,
+) -> Result<()> {
+    let state = &mut ctx.accounts.state;
+
+    if !ctx.accounts.route.enabled {
+        return err!(CommonError::DisabledRoute);
+    }
```

Comment:
This is redundant as this uses DepositV3 context that already checked the route.

Reply:
and we better not reimplement the deposit_v3 logic here, but call it internally